### DISCUSSION TITLE
nix support for emacs' config

### DISCUSSION THREAD

**Discussion by bbigras** [emacs-lsp/lsp-gitpod#20] [Ideas]
Some nix users are using [programs.emacs](https://rycee.gitlab.io/home-manager/options.html#opt-programs.emacs.enable) to configure emacs. It would be nice to be able to support it, so people could use the same emacs config that on their other machines.

In my case, I would like to be able to use the [emacs-int home-manager module](https://gitlab.com/rycee/nur-expressions/blob/master/hm-modules/emacs-init.nix), so I could configure emacs like this: https://github.com/bbigras/nix-config/blob/1918b24299962feaca5e0006764871e1362fca44/users/bbigras/dev/emacs.nix

**Comment by yyoncho** [comment]
ATM gitpod supports ubuntu/debian based docker images. As long as this can run on top of them, then it is possible. I am not very familiar with nix stuff. @ericdallo, any thoughts?

**Reply by ericdallo** [reply]
@yyoncho it's possible to install `home-manager` (a Nix program to install/setup nix derivations for a user) on debian baseds, it should work I think

**Comment by yyoncho** [comment]
Based on @ericdallo the solution is:

1. Create a new docker image based on yyoncho/gitpod-default . Use your nix files on it and use `home-manager`
2. Push your docker image in docker hub
3. Then in your dot files repo place 
```
.gitpod.Dockerfile
.gitpod.yml
```
and fix `.gitpod.Dockerfile` to point to the image that you have created.

**Reply by ericdallo** [reply]
If that works we can later think in more user friendly way, but that looks enough for now

**Reply by yyoncho** [reply]
This is the best way. Otherwise,  you will have to wait for the installation and configuration process when you start the workspace... What we can provide is a template, like put your nix config then do this and that.
